src/h5web/metadata-viewer/AttributesTable.tsx

```diff
@@ -31,14 +37,18 @@ function AttributesTable(props: Props) {
         </tr>
       </thead>
       <tbody>
-        {attributes.map(({ name, value }) => (
+        {attributes.map(({ name, type, value }) => (
           <tr key={name}>
             <th scope="row">{name}</th>
             <td>
               {FOLLOWABLE_ATTRS.has(name) ? (
                 <AttributeValueLink onFollowPath={onFollowPath} value={value} />
               ) : (
-                JSON.stringify(value)
+                JSON.stringify(
+                  type.class === DTypeClass.Complex
+                    ? formatComplex(value as H5WebComplex | ComplexArray)
+                    : value
```

Comment:
We can surely avoid the cast here using a type guard. My issue is: what args does this type guard take ? 

I can't give the whole `attribute` as TS does not allow me to infer the type of an object field. But it would seem awkward to give `type` and `value`, no ?

Reply:
Instinctively, it looks like the same problem as with datasets. We'd first want a guard that checks whether the attribute has a complex type and a scalar or array shape (`Attribute<Shape, ComplexType>`), then we'd want `formatComplex` to accept the entire attribute object so that the signature of the utility function can make use of `Value<Dataset | Attribute>`.

I wonder whether we could make all our existing guards smarter so they could deal with both datasets and attributes (with an extra generic).